source/features/batch-mark-files-as-viewed.tsx

```diff
@@ -38,10 +40,33 @@ function batchToggle(event: delegate.Event<MouseEvent, HTMLFormElement>): void {
 	}
 }
 
+function markAsViewedSelector(target: HTMLElement): string {
+	const checked = (target as HTMLInputElement).checked ? ':not([checked])' : '[checked]';
+	return '.js-reviewed-checkbox' + checked;
+}
+
+const markAsViewed = clickAll(markAsViewedSelector);
+
+function onAltClick(event: delegate.Event<MouseEvent, HTMLInputElement>): void {
+	if (!event.altKey || !event.isTrusted) {
+		return;
+	}
+
+	void showToast(async () => {
+		markAsViewed(event);
+	}, {
+		message: event.delegateTarget.checked ?
+			'Marking visible files as viewed' :
+			'Marking visible files as unviewed',
+		doneMessage: 'Marking files completed'
+	});
+}
+
 function init(): void {
 	// `mousedown` required to avoid mouse selection on shift-click
 	delegate(document, '.js-toggle-user-reviewed-file-form', 'mousedown', batchToggle);
 	delegate(document, '.js-toggle-user-reviewed-file-form', 'submit', remember);
+	delegate(document, '.js-reviewed-checkbox', 'click', onAltClick);
```

Comment:
@yakov116 Could this be extended to also allow alt-clicking on the label?

The selector would then be `.js-reviewed-toggle`, but in `onAltClick`, the checkbox needs to be selected first.